### DISCUSSION TITLE
read_value not updating, unless subscription is created

### DISCUSSION THREAD

**Discussion by camialbarracinc** [FreeOpcUa/opcua-asyncio#818] [Q&A]
Hi!

I've just started using the library a few days ago, built my own server and client using the examples and tested different methods, everything was going smoothly until I tried to connect a client built with the library to a Movicon OPC UA server. 

When I try to perform var.read_value() it shows the value of the variable but if the variable changes on the server and I perform read_value() again It doesn't show the updated value it shows the original value.
However If I create a subscription on datachange to the variable It does show the correct value every time I perform read_value.

This was not the behaviour I was experiencing when using the included server class, with the included server if I changed the value of the variable it would show the updated value on the client every time I called read_value without the need of creating a subscription.

I was wondering is this a normal behaviour?, It's due to the Movicon server?, Do I really have to create a subscription for each variable I want to retrieve with read_value? and are those subscription generating more load on the client/server?

Thank you in advance!

**Comment by AndreasHeine** [comment]
have you tried to browse the variable before reading it!? had it with some products that they start updating from the subsystem only if there was a initial browse on the variable...

**Comment by schroeder-** [comment]
Have you tried it with some other client like UAExpert? Is the server behavior the same?

**Comment by camialbarracinc** [comment]
Thank you for your responses!

@AndreasHeine I'm not sure what you're referring as browsing the variable basically I'm performing this two operations
> var = await client.nodes.root.get_child(["0:Objects", f"{idx}:Tags", f"{idx}:CV", f"{idx}:CV1"])
> received_signal = await var.read_value()

@schroeder- Yes I've tried with the client included in Movicon and I do see de updates in the client as soon as they're updated.

**Reply by AndreasHeine** [reply]
@camialbarracinc the opc ua server has some sort of subsystem / data source where the server gets data from and some products implement some weird behavior like they start sampling the data source in the subsystem only if the variable is actively used... there must be a trigger which forces the opc ua server for start sampling the source (in some its browse them once before using it)!

**Reply by camialbarracinc** [reply]
Ohh, I wasn't aware of this behaviour, wouldn't it be enough to perform one(or several) read_value() operations so the server start sampling the source?, Or how have you performed this browse operation?.

I'll start digging in the OPC Server properties and options to see If I can find something that relates to this behaviour.

**Reply by swamper123** [reply]
I would guess that reading is an option (if the server lets you).
Other option I would use is `get_child`:

`await client.nodes.root.get_child(["0:Objects", f"{idx}:MyObject", f"{idx}:MyVariable"])`
